flink/v1.12/flink/src/main/java/com/netease/arctic/flink/util/ArcticUtils.java

```diff
@@ -177,17 +196,58 @@ public static ArcticLogWriter buildArcticLogWriter(Map<String, String> propertie
         "'. only support 'v1' or empty");
   }
 
-  public static <T> LogMsgFactory<T> buildLogMsgFactory(Map<String, String> tableProperties) {
+  private static Properties combineArcticProperties(Map<String, String> tableProperties,
+                                                    Properties producerConfig) {
+    Properties finalProp;
+    Properties props = getLogStoreProperties(tableProperties);
+    if (producerConfig == null) {
+      finalProp = props;
+    } else {
+      props.stringPropertyNames().forEach(k -> producerConfig.putIfAbsent(k, props.get(k)));
+      finalProp = producerConfig;
+    }
+    
+    String logStoreAddress = CompatibleFlinkPropertyUtil.propertyAsString(tableProperties,
+        LOG_STORE_ADDRESS, null);
+    
```

Comment:
```java
Preconditions.checkArgument(logStoreAddress, String.format("%s should be set",
          LOG_STORE_ADDRESS));
```
It is better to put validation here and remove the below redundant empty variable check.

Reply:
I think logStoreAddress can be set by bootstrap.servers or pulsar.client.serviceUrl directly, then LOG_STORE_ADDRESS can be unnecessary.